src/libmongoc/tests/test-mongoc-handshake.c

```diff
@@ -276,6 +276,122 @@ test_mongoc_handshake_data_append_success (void)
 }
 
 
+static void
+test_mongoc_handshake_data_append_null_args (void)
+{
+   mock_server_t *server;
+   mongoc_uri_t *uri;
+   mongoc_client_t *client;
+   mongoc_client_pool_t *pool;
+   request_t *request;
+   const bson_t *request_doc;
+   bson_iter_t iter;
+   bson_iter_t md_iter;
+   bson_iter_t inner_iter;
+   const char *val;
+
+   char big_string[HANDSHAKE_MAX_SIZE];
+
+   memset (big_string, 'a', HANDSHAKE_MAX_SIZE - 1);
+   big_string[HANDSHAKE_MAX_SIZE - 1] = '\0';
+
+   _reset_handshake ();
+   /* Make sure setting the handshake works */
+   ASSERT (mongoc_handshake_data_append (NULL, NULL, NULL));
+
+   server = mock_server_new ();
+   mock_server_run (server);
+   uri = mongoc_uri_copy (mock_server_get_uri (server));
+   mongoc_uri_set_option_as_int32 (uri, MONGOC_URI_HEARTBEATFREQUENCYMS, 500);
```

Comment:
I suggest removing L305. I see `MONGOC_URI_HEARTBEATFREQUENCYMS` is set in `test_mongoc_handshake_data_append_success`. `test_mongoc_handshake_data_append_success` looks like the template of this test. But setting `MONGOC_URI_HEARTBEATFREQUENCYMS` is not necessary.

The test needs to observe the MongoDB handshake from server monitoring. `MONGOC_URI_HEARTBEATFREQUENCYMS` determines the frequency server monitoring checks. But the first check will occur shortly after the call to `mongoc_client_pool_pop`.

Reply:
Should the line be removed from `test_mongoc_handshake_data_append_success()` as well?